File: MVVMFrame/src/main/java/com/leonyr/lib/mvvm/act/AbBindActivity.java
package com.leonyr.lib.mvvm.act;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;

/**
 * ==============================================================
 * Description: databing activity 基类
 * <p>
 * Created by leonyr on 2019-04-20
 * (C) Copyright LeonyR Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
public abstract class AbBindActivity<T extends ViewDataBinding> extends AppCompatActivity {

    protected String TAG;
    protected Context mCtx;
    private T binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        super.onCreate(savedInstanceState);
        getLifecycle().addObserver(new ActivityObserver());

        mCtx = this;
        binding = DataBindingUtil.setContentView(this, getLayoutResId());
        TAG = getClass().getSimpleName();

        initView(savedInstanceState);
    }

    public T Binding() {
        return binding;
    }

    @Override
    protected void onResume() {
        super.onResume();
        TAG = getClass().getSimpleName();
    }

    @Override
    public void onDestroy() {
        mCtx = null;
        super.onDestroy();
    }

    /**
     * 获取layout id
     */
    @LayoutRes
    protected abstract int getLayoutResId();

    /**
     * 初始化其他数据Data
     */
    protected abstract void initView(Bundle savedInstanceState);

}

File: MVVMFrame/src/main/java/com/leonyr/mvvm/act/AbBindActivity.kt
package com.leonyr.mvvm.act

import android.content.Context
import android.os.Bundle
import android.view.Window
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.Observer
import com.leonyr.mvvm.net.NetLoading

import com.leonyr.mvvm.vm.LViewModel

/**
 * ==============================================================
 * Description: databing activity 基类
 *
 *
 * Created by leonyr on 2019-04-20
 * (C) Copyright LeonyR Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
abstract class AbBindActivity<VM : LViewModel, T : ViewDataBinding> : AppCompatActivity() {

    lateinit var dialogLoading: NetLoading

    lateinit var TAG: String
    lateinit var mCtx: Context
    lateinit var binding: T

    lateinit var vModel: VM

    /**
     * 获取layout id
     */
    @get:LayoutRes
    protected abstract val layoutResId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        super.onCreate(savedInstanceState)
        lifecycle.addObserver(ActivityObserver())

        mCtx = this
        dialogLoading = NetLoading(mCtx)
        binding = DataBindingUtil.setContentView(this, layoutResId)
        TAG = javaClass.simpleName

        initView(savedInstanceState)
    }

    fun observerLoading() {
        vModel.showLoading.observe(this, Observer { show ->
            if (show) {
                dialogLoading.show()
            } else {
                dialogLoading.dismiss()
            }
        })
    }

    override fun onResume() {
        super.onResume()
        TAG = javaClass.simpleName

    }

    /**
     * 初始化其他数据Data
     */
    protected abstract fun initView(savedInstanceState: Bundle?)

    /*override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        // Forward results to EasyPermissions
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
    }*/

}
